### PR TITLE
Adds retry documentation and demo

### PR DESCRIPTION
**Pull request by shalier**
Adds documentation and demo for retry policy and fixes typo
resolves https://github.com/openservicemesh/osm-docs/issues/318

### CODE REVIEW

**Comment by shalier** [comment]
In OSM v1.1.1 the Prometheus config does not contain the envoy retry metrics. Hold merge until the config has the metrics, because the demo relies on them

**Review by nojnhuh — CHANGES_REQUESTED** [review]
Overall looks good, just left a few comments for small stuff.

> In OSM v1.1.1 the Prometheus config does not contain the envoy retry metrics. Hold merge until the config has the metrics, because the demo relies on them

If the metrics are already included in `main` then that shouldn't block this PR since the live version of the docs are built from the release branch so the `main` branch here can stay in sync with the `main` branch of the code repo.

File: content/docs/demos/retry_policy.md [hunk 1]
```diff
@@ -0,0 +1,120 @@
+---
+title: "Retry Policy"
+description: "Set up application with retry"
```
Comment (shashankram) [inline content/docs/demos/retry_policy.md:3]:
```suggestion
description: "Using retries to enhance service availability"
```

File: content/docs/demos/retry_policy.md [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+    Confirm the `curl` pod is up and running.
+
+    ```console
+    $ kubectl get pods -n curl
+    NAME                    READY   STATUS    RESTARTS   AGE
+    curl-54ccc6954c-9rlvp   2/2     Running   0          20s
+    ```
+
+1. Apply the Retry policy to retry when the `curl` ServiceAccount receives a `5xx` code when sending a request to `httpbin` Service.
+```bash
```
Comment (shashankram) [inline content/docs/demos/retry_policy.md:66]: This an additional indent for correct formatting.

**Review by shashankram — COMMENTED** [review]
Nice doc @shalier

File: content/docs/demos/retry_policy.md [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+kubectl apply -f - <<EOF
+kind: Retry
+apiVersion: policy.openservicemesh.io/v1alpha1
+metadata:
+  name: retry
```
[the remark this answers is not in the document]
Comment (shalier) [reply content/docs/demos/retry_policy.md]: The retry policy should still work even if it's in the `default` namespace, lmk if it doesn't work for you

**Review by shashankram — APPROVED** [review]

File: content/docs/demos/retry_policy.md [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+
+1. Apply the Retry policy to retry when the `curl` ServiceAccount receives a `5xx` code when sending a request to `httpbin` Service.
+    ```bash
+    kubectl apply -f - <<EOF
+    kind: Retry
+    apiVersion: policy.openservicemesh.io/v1alpha1
+    metadata:
+    name: retry
+    spec:
+    source:
```
Comment (nojnhuh) [inline content/docs/demos/retry_policy.md:76]: Looks like the indentation here is a bit off.

**Review by nojnhuh — APPROVED** [review]

**Pull request merged**